gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -2269,6 +2271,32 @@ public default GraphTraversal<S, E> drop() {
         return this.asAdmin().addStep(new DropStep<>(this.asAdmin()));
     }
 
+    /**
+     * Filters <code>E</code> lists given the provided {@code predicate}.
+     *
+     * @param predicate the filter to apply
+     * @return the traversal with an appended {@link AllStep}
+     * @see <a href="http://tinkerpop.apache.org/docs/${project.version}/reference/#all-step" target="_blank">Reference Documentation - All Step</a>
+     * @since 3.0.0-incubating
+     */
+    public default <S2> GraphTraversal<S, E> all(final P<S2> predicate) {
```

Comment:
is `all` step return data type same as input data type? S==E?

Reply:
Yes, they are the same. Did you prefer that I can to just S without E?